mne/io/nsx/nsx.py

```diff
@@ -0,0 +1,562 @@
+            pick_mask[idx] = False
+        elif idx in stim_channel_idxs:
+            chan_info["coil_type"] = FIFF.FIFFV_COIL_NONE
+            chan_info["unit"] = FIFF.FIFF_UNIT_NONE
+            chan_info["kind"] = FIFF.FIFFV_STIM_CH
+            pick_mask[idx] = False
+            chan_info["ch_name"] = ch_name
+            ch_names[idx] = chan_info["ch_name"]
+        chs.append(chan_info)
+
+    sfreq = nsx_info["timestamp_resolution"] / nsx_info["period"]
+    info = _empty_info(sfreq)
+    info["meas_date"] = nsx_info["meas_date"]
+    info["chs"] = chs
+    info["ch_names"] = ch_names
+
+    highpass = nsx_info["highpass"][:128]
+    lowpass = nsx_info["lowpass"][:128]
+    _decode_online_filters(info, highpass, lowpass)
+
+    # Some keys to be consistent with FIF measurement info
+    info["description"] = None
+
+    info._unlocked = False
+    info._update_redundant()
+
+    orig_format = ORIG_FORMAT
+
+    raw_extras = {
+        key: [r[key] for r in nsx_info["data_header"]]
+        for key in nsx_info["data_header"][0]
+    }
+    for key in raw_extras:
+        raw_extras[key] = np.array(raw_extras[key], int)
+    good_data_packets = raw_extras.pop("header") == 1
+    if not good_data_packets.any():
+        raise RuntimeError("NSx file appears to be broken")
+    raw_extras = {key: raw_extras[key][good_data_packets] for key in raw_extras.keys()}
+    raw_extras["timestamp"] = raw_extras["timestamp"] // nsx_info["period"]
+    first_samp = raw_extras["timestamp"][0]
+    last_samp = raw_extras["timestamp"][-1] + raw_extras["nb_data_points"][-1]
+    n_samples = last_samp - first_samp
+
+    return (
+        info,
+        fname,
+        nsx_info["spec"],
+        n_samples,
+        orig_format,
+        raw_extras,
+        orig_units,
+    )
+
+
+def _decode_online_filters(info, highpass, lowpass):
+    """Decode low/high-pass filters that are applied online."""
+    if np.all(highpass == highpass[0]):
+        if highpass[0] == "NaN":
+            # Placeholder for future use. Highpass set in _empty_info.
+            pass
+        else:
+            hp = float(highpass[0])
+            info["highpass"] = hp
+    else:
+        info["highpass"] = float(np.max(highpass))
+        warn(
+            "Channels contain different highpass filters. Highest filter "
+            "setting will be stored."
+        )
+
+    if np.all(lowpass == lowpass[0]):
+        if lowpass[0] in ("NaN", "0", "0.0"):
+            # Placeholder for future use. Lowpass set in _empty_info.
+            pass
+        else:
+            info["lowpass"] = float(lowpass[0])
+    else:
+        info["lowpass"] = float(np.min(lowpass))
+        warn(
+            "Channels contain different lowpass filters. Lowest filter "
+            "setting will be stored."
+        )
+
+
+def _check_stim_channel(stim_channel, ch_names):
+    """Check that the stimulus channel exists in the current datafile."""
+    DEFAULT_STIM_CH_NAMES = ["status", "trigger"]
```

Comment:
Is this done elsewhere in MNE? Would be nice to reuse the global variable so that it's guaranteed to be consistent across file types

Reply:
I think this is file type dependent, so we would not want to make it a global variable.

Reply:
Yes, for sure but these are the same names as are used elsewhere so it would be nice to deduplicate. Maybe put a triggers channel name dictionary in defaults.py would be helpful.

Reply:
I might do this as a followup PR, it would be nice to have a list of reserved channel names per file type in one place so you don't accidentally use them etc. but I get what you mean that this is the nsx format spec so looks good for this PR.